DMCompiler/DM/Expressions/Builtins.cs

```diff
@@ -600,7 +600,22 @@ sealed class Nameof : DMExpression {
         }
 
         public override void EmitPushValue(DMObject dmObject, DMProc proc) {
-            proc.PushString(_expr.GetNameof(dmObject, proc));
+            if (_expr.GetNameof() is { } nameof) {
+                proc.PushString(nameof);
+            } else {
+                DMCompiler.Emit(WarningCode.BadArgument, Location, "nameof() requires a var, proc reference, or type path");
+                proc.PushString("");
+            }
+        }
+
+        public override bool TryAsConstant([NotNullWhen(true)] out Constant? constant) {
+            if (_expr.GetNameof() is { } nameof) {
```

Comment:
Instead of copy/pasting this code, it would be better to perform this once in the constructor and save the result in a `private string _nameof`.

Reply:
Though feel free to ignore this if you're going to re-add the dmObject and proc args, since that wouldn't be possible.